include/envoy/network/address.h

```diff
@@ -125,5 +125,104 @@ class Instance {
 
 typedef std::shared_ptr<const Instance> InstancePtr;
 
+/*
+ * Parse an internet host address (IPv4 or IPv6) and create an Instance from it.
+ * The address must not include a port number.
+ * @param ipAddr string to be parsed as an internet address.
+ * @return pointer to the Instance, or nullptr if unable to parse the address.
+ */
+InstancePtr parseInternetAddress(const std::string& ipAddr);
```

Comment:
There isn't. Sorry. :( I will write something up. This isn't a huge deal and I'm open to changing things up, but in general so far we have been putting abstract interfaces in include/ (that generally map to what is in mocks), except for a few small exceptions generally related to perf requirements.

So basically include/ (abstract cross component interfaces that need mocking)
source/ (interface implementations and "simple" classes that don't have interfaces)